packages/token/src/tests/erc20/test_erc20.cairo

```diff
@@ -449,3 +451,135 @@ fn test_burn_from_zero() {
+
+    let mut spy = spy_events();
+    let contract_address = test_address();
+
+    start_cheat_caller_address(contract_address, OWNER());
+
+    assert_state_before_transfer(OWNER(), RECIPIENT());
+    state.update(OWNER(), RECIPIENT(), VALUE);
+    assert_state_after_transfer(OWNER(), RECIPIENT(), VALUE);
+
+    spy.assert_only_event_transfer(contract_address, OWNER(), RECIPIENT(), VALUE);
+}
+
+#[test]
+fn test_update_from_non_zero_to_zero() {
+    let mut state = setup();
+
+    let mut spy = spy_events();
+    let contract_address = test_address();
+
+    start_cheat_caller_address(contract_address, OWNER());
+
+    assert_state_before_burn(OWNER());
+    state.update(OWNER(), ZERO(), VALUE);
+    assert_state_after_burn(OWNER(), VALUE);
+
+    spy.assert_only_event_transfer(contract_address, OWNER(), ZERO(), VALUE);
+}
+
+#[test]
+fn test_update_from_zero_to_non_zero() {
+    let mut state = setup();
+
+    let mut spy = spy_events();
+    let contract_address = test_address();
+
+    start_cheat_caller_address(contract_address, RECIPIENT());
+
+    assert_state_before_mint(RECIPIENT());
+    state.update(ZERO(), RECIPIENT(), VALUE);
+    assert_state_after_mint(RECIPIENT(), VALUE);
+
+    spy.assert_only_event_transfer(contract_address, ZERO(), RECIPIENT(), VALUE);
+}
+
+#[test]
+fn test_update_from_zero_to_zero() {
+    // The update function should not be called from zero to zero
+    // because this does nothing to the state.
+    let mut state = setup();
+
+    let mut spy = spy_events();
+    let contract_address = test_address();
+    let supply_before_update = state.total_supply();
+
+    start_cheat_caller_address(contract_address, RECIPIENT());
+
+    state.update(ZERO(), ZERO(), VALUE);
+
+    let supply_after_update = state.total_supply();
+    assert_eq!(supply_before_update, supply_after_update);
+
+    spy.assert_only_event_transfer(contract_address, ZERO(), ZERO(), VALUE);
+}
+
+//
+// Helpers
+//
+
+fn assert_state_before_transfer(sender: ContractAddress, recipient: ContractAddress) {
+    let state = COMPONENT_STATE();
+    let initial_supply = SUPPLY;
+    let current_supply = state.total_supply();
+
+    assert_eq!(initial_supply, current_supply);
+    assert_eq!(state.balance_of(sender), SUPPLY);
+    assert!(state.balance_of(recipient).is_zero());
+}
+
+fn assert_state_after_transfer(sender: ContractAddress, recipient: ContractAddress, amount: u256) {
+    let state = COMPONENT_STATE();
+    let initial_supply = SUPPLY;
+    let current_supply = state.total_supply();
+
+    assert_eq!(initial_supply, current_supply);
+    assert_eq!(state.balance_of(sender), initial_supply - amount);
+    assert_eq!(state.balance_of(recipient), amount);
+}
+
+fn assert_state_before_mint(recipient: ContractAddress) {
+    let state = COMPONENT_STATE();
+    let initial_supply = SUPPLY;
+    let current_supply = state.total_supply();
+
+    assert_eq!(current_supply, initial_supply);
+    assert!(state.balance_of(recipient).is_zero());
```

Comment:
```suggestion
    assert!(state.balance_of(recipient) == 0);
```
That will make it easier to debug

Reply:
Mmm agreed, but it's probably better to use assert_eq with `0`